### PR TITLE
User modal role error

### PR DESCRIPTION
**Pull request by canaandavis**

### CODE REVIEW

File: public/src/js/components/userModalComponent.js [hunk 1]
```diff
@@ -7,107 +7,222 @@ var UserModel = require('../models/userModel');
...
-    'is_instructor',
-    'is_student'
-  ],
-
-  componentDidMount: function() {
-    _.each(this.attributes, function (attr) {
-      this.refs[attr].value = this.props.model.get(attr);
-    }, this);
+
+  getInitialState: function() {
```
Comment (mistakevin) [inline public/src/js/components/userModalComponent.js]: Now that I think of it, the library already creates the state based on the model attributes. You can access them at `this.state.model`. The main purpose of the library is to just glue together `this.state.model` and `this.props.model.attributes`. So we may not need to use `getInitialState()`

Comment (canaandavis) [inline public/src/js/components/userModalComponent.js]: I think this approach can work.  The only real "gotcha", I can see at the moment, would be we have to look out for is overriding computed attributes on the model.

On the userModel something like full_name, would be overridden if we were to simply save the model with this.state.model. We would either be saving with an empty default, or an out-of-sync version.  Either will overwrite the calculated version that is generated in the userModel, if the key full_name in this.state.model comes after the first_name and last_name keys.

Check out my updated PR, In the saveUser function I'm being explicit in drawing out only the attributes we want to include.  

We could switch this up and remove the computed attributes, but I think it is less clear what is actually going on that way.

File: public/src/js/components/userModalComponent.js [hunk 2]
```diff
@@ -7,107 +7,187 @@ var UserModel = require('../models/userModel');
...
     var that = this;
-    
-    var attrs = {};
-    _.each(this.attributes, function(attr) {
-      attrs[attr] = that.refs[attr].value
-    });
-    
-    this.props.model.save(attrs, {
+
+    this.props.model.save(this.state.model, {
```
Comment (canaandavis) [inline public/src/js/components/userModalComponent.js]: We can pass in this.state.model here

File: public/src/js/models/userModel.js [hunk 3]
```diff
@@ -17,16 +17,16 @@ module.exports = Backbone.Model.extend({
     last_name: '',
     full_name: ''
   },
-  
+
   initialize: function() {
     this.fullName();
-    this.on('change:first_name change:last_name', this.fullName);
+    this.on('change:first_name change:last_name sync', this.fullName);
```
Comment (canaandavis) [inline public/src/js/models/userModel.js]: Listen for a sync here to re-calculate the computed attributes

**Comment by mistakevin** [comment]
I see what you did there. Looks good to merge.

**Comment by mistakevin** [comment]
resolves https://github.com/mistakevin/aca-campus/issues/1

**Pull request merged**